### ISSUE TITLE
nextjs project's hot reloading is malfunctioning

### ISSUE THREAD

**Issue by jaketakula** [nrwl/nx#20328]
### Current Behavior

hot reloading failed with sample nextjs project.

### Expected Behavior

hot reloading failed with sample nextjs project shoudl work

### Steps to Reproduce

nvm use 18
npm install -g pnpm

pnpx create-nx-workspace@16.9.1 wkspace --workspaceType=integrated \
--preset=apps --nxCloud=false -y

pnpx nx@16.9.1 g @nx/next:app  helloworld --style=@emotion/styled --appDir=true --directory=apps \
--e2eTestRunner=playwright --swc=true

pnpx nx@16.9.1 run cloud-ui:serve --configuration=development

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 18.18.0
   OS     : darwin-x64
   pnpm   : 8.8.0
   
   nx (global)        : 16.8.1
   nx                 : 16.9.1
   @nx/js             : 16.9.1
   @nrwl/js           : 16.7.4
   @nx/jest           : 16.7.4
   @nx/linter         : 16.7.4
   @nx/workspace      : 16.9.1
   @nrwl/workspace    : 16.7.4
   @nx/devkit         : 16.7.4
   @nx/eslint-plugin  : 16.7.4
   @nx/nest           : 16.7.4
   @nx/next           : 16.7.4
   @nx/node           : 16.7.4
   @nx/playwright     : 16.7.4
   @nx/react          : 16.7.4
   @nrwl/tao          : 16.7.4
   @nx/web            : 16.7.4
   typescript         : 5.1.6
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/js@16.7.4
     - @nx/jest@16.7.4
     - @nrwl/jest@16.7.4
     - @nx/linter@16.7.4
     - @nrwl/linter@16.7.4
     - @nrwl/workspace@16.7.4
     - @nx/devkit@16.7.4
     - @nrwl/devkit@16.7.4
     - @nx/eslint-plugin@16.7.4
     - @nrwl/eslint-plugin-nx@16.7.4
     - @nx/nest@16.7.4
     - @nrwl/nest@16.7.4
     - @nx/next@16.7.4
     - @nrwl/next@16.7.4
     - @nx/node@16.7.4
     - @nrwl/node@16.7.4
     - @nx/playwright@16.7.4
     - @nx/react@16.7.4
     - @nrwl/react@16.7.4
     - @nrwl/tao@16.7.4
     - @nx/web@16.7.4
     - @nrwl/web@16.7.4
   
   To fix this, run `nx migrate nx@16.9.1`
```

### Failure Logs

```shell
Failure Logs
Executing task: pnpm exec nx run cloud-ui:serve

nx run cloud-ui:serve:development

ready started server on 0.0.0.0:4200, url: http://localhost:4200
event compiled client and server successfully in 1904 ms (311 modules)
wait compiling...
event compiled client and server successfully in 1295 ms (311 modules)
wait compiling /page (client and server)...
warn ../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js
Attempted import error: 'createContext' is not exported from 'react' (imported as 'React').
Import trace for requested module:
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/jsx-dev-runtime/dist/emotion-react-jsx-dev-runtime.esm.js
./app/page.tsx
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js
Attempted import error: 'createContext' is not exported from 'react' (imported as 'React').
Import trace for requested module:
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/jsx-dev-runtime/dist/emotion-react-jsx-dev-runtime.esm.js
./app/page.tsx
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-react.esm.js
Attempted import error: 'useRef' is not exported from 'react' (imported as 'React').
Import trace for requested module:
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-react.esm.js
../../node_modules/.pnpm/@emotion+styled@11.11.0_@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/styled/dist/emotion-styled.esm.js
./app/page.tsx
../../node_modules/.pnpm/@emotion+use-insertion-effect-with-fallbacks@1.0.1_react@18.2.0/node_modules/@emotion/use-insertion-effect-with-fallbacks/dist/emotion-use-insertion-effect-with-fallbacks.esm.js
Attempted import error: 'useLayoutEffect' is not exported from 'react' (imported as 'React').
Import trace for requested module:
../../node_modules/.pnpm/@emotion+use-insertion-effect-with-fallbacks@1.0.1_react@18.2.0/node_modules/@emotion/use-insertion-effect-with-fallbacks/dist/emotion-use-insertion-effect-with-fallbacks.esm.js
../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/jsx-dev-runtime/dist/emotion-react-jsx-dev-runtime.esm.js
./app/page.tsx
error ../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js (14:41) @ React
error createContext only works in Client Components. Add the "use client" directive at the top of the file to use it. Read more: https://nextjs.org/docs/messages/context-in-server-component
at eval (webpack-internal:///(sc_server)/../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js:34:92)
at (sc_server)/../../node_modules/.pnpm/@emotion+react@11.11.1_@types+react@18.2.14_react@18.2.0/node_modules/@emotion/react/dist/emotion-element-6bdfffb2.esm.js (/Users/jakzha/Documents/takula-nita/apps/cloud-ui/.next/server/app/page.js:1636:1)
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

https://github.com/nrwl/nx/issues/19399
I reported this issue in this old post.
but it was confirmed not nx issue.
but the sample nextjs created by nx command did Not work by default.
I thinkl it still need to be fixed by nx team.

**Comment by erhanerdgn** [comment]
I am facing this problem right now.

The next@v14.0.5-canary.19 version did not have this problem, but v14.0.5-canary.20 and all updates above do not work with NX.

When we try the same project with **next dev** or **next start**, it works without any problems.

NX Report
```
Node   : 20.10.0
OS     : darwin-arm64
pnpm   : 8.12.1
   
nx (global)        : 17.2.7
nx                 : 17.2.7
@nx/js             : 17.2.7
@nx/jest           : 17.2.7
@nx/eslint         : 17.2.7
@nx/workspace      : 17.2.7
@nx/eslint-plugin  : 17.2.7
@nx/next           : 17.2.7
@nx/node           : 17.2.7
@nx/react          : 17.2.7
@nx/storybook      : 17.2.7
@nrwl/tao          : 17.2.7
@nx/web            : 17.2.7
typescript         : 5.3.3
```

Next.js Info
```
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:53:34 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T8103
Binaries:
  Node: 20.10.0
  npm: 10.2.3
  Yarn: N/A
  pnpm: 8.12.1
Relevant Packages:
  next: 14.0.5-canary.20
  eslint-config-next: 14.0.5-canary.20
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

**Comment by ndcunningham** [comment]
Can you try this without using `@emotion` style? 
The support for emotion is still being worked on: https://nextjs.org/docs/app/building-your-application/styling/css-in-js
I tried this and it is working as expected.
Here is a repo: https://github.com/ndcunningham/issue-20328

**Comment by erhanerdgn** [comment]
It's not about my emotion or any style.

As soon as there is any Provider added on layout.tsx, it gives hooks errors in the same way on whatever that plugin is. 

Ex; react-query, nextUI, reCaptchaProvider and similar..

```
import '../../styles/global.css';

import type { Viewport } from 'next';
import { Inter } from 'next/font/google';
import { ReCaptchaProvider } from 'next-recaptcha-v3';

import { UIProvider } from '../UIProvider';

import { StoreProviders } from '@/store/provider';

const inter = Inter({
  weight: ['400', '500', '700'],
  style: ['normal'],
  subsets: ['latin'],
  variable: '--font-inter',
  display: 'swap'
});

export const viewport: Viewport = {
  width: 'device-width',
  initialScale: 1,
  maximumScale: 1,
  viewportFit: 'cover',
  userScalable: false
};

export default function RootLayout({
  children,
  params: { locale }
}: {
  children: React.ReactNode;
  params: { locale: string };
}): React.ReactNode {
  return (
    <html lang={locale} className={inter.variable} suppressHydrationWarning>
      <body className={process.env.NODE_ENV === 'development' ? 'debug-screens' : ''}>
        <StoreProviders>
          <UIProvider>
            <ReCaptchaProvider reCaptchaKey="[6LfMq3sbAAAAAIhVxgwqrHZ4VRTBciqGUDOxvWYo]" language={locale}>
              {children}
            </ReCaptchaProvider>
          </UIProvider>
        </StoreProviders>
      </body>
    </html>
  );
}
```

Error 1:
```
Attempted import error: 'forwardRef' is not exported from 'react' (imported as 'baseForwardRef').
Import trace for requested module:
../../node_modules/.pnpm/@nextui-org+system-rsc@2.0.11_@nextui-org+theme@2.1.17_react@18.2.0_tailwind-variants@0.1.19/node_modules/@nextui-org/system-rsc/dist/chunk-SF6SJHAM.mjs
../../node_modules/.pnpm/@nextui-org+system-rsc@2.0.11_@nextui-org+theme@2.1.17_react@18.2.0_tailwind-variants@0.1.19/node_modules/@nextui-org/system-rsc/dist/index.mjs
../../node_modules/.pnpm/@nextui-org+system@2.0.15_@nextui-org+theme@2.1.17_react-dom@18.2.0_react@18.2.0_tailwind-variants@0.1.19/node_modules/@nextui-org/system/dist/index.mjs
./app/UIProvider.tsx
```

Error 2:
```
Attempted import error: 'useRef' is not exported from 'react' (imported as 'React').
Import trace for requested module:
../../node_modules/.pnpm/@tanstack+react-query-devtools@5.15.0_@tanstack+react-query@5.15.0_react@18.2.0/node_modules/@tanstack/react-query-devtools/build/modern/devtools.js
../../node_modules/.pnpm/@tanstack+react-query-devtools@5.15.0_@tanstack+react-query@5.15.0_react@18.2.0/node_modules/@tanstack/react-query-devtools/build/modern/index.js
../../libs/store/src/lib/provider.tsx
```

Error 3:
```
Attempted import error: 'useEffect' is not exported from 'react' (imported as 'useEffect').
Import trace for requested module:
../../node_modules/.pnpm/next-recaptcha-v3@1.3.0_next@14.0.5-canary.29_react-dom@18.2.0_react@18.2.0/node_modules/next-recaptcha-v3/lib/ReCaptcha.js
```

**Comment by erhanerdgn** [comment]
@ndcunningham, Do you have any feedback?

**Comment by nej1044** [comment]
I am also facing the same problem.

The nx team said that this problem is due to Next.js not supporting the css-in-js library(#19399, #19224, #16627)
The problem occurs in components (app/layout.tsx) that do not use css-in-js.

Is there any solution to this problem?

**Comment by jaketakula** [comment]
+1 to kkep this issue open.
i really want to contribute but i do not have time and efforts to fet into the nx impl.
can you guys help ?

**Comment by erhanerdgn** [comment]
It seems to be fixed with v14.0.5-canary.67

**Comment by jaketakula** [comment]
this bug is fixed in the recent release of 17.3.0. Thx  very much. closed.
